server/src/main/java/org/opensearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -748,7 +748,7 @@ private static DiscoveryNode findSourceNodeForPeerRecovery(
         return sourceNode;
     }
 
-    private class ReplicationListener implements SegmentReplicationReplicaService.ReplicationListener {
+    private class ShardRoutingReplicationListener implements SegmentReplicationReplicaService.SegmentReplicationListener {
```

Comment:
This was me being quick and dirty with the POC - I don't think we need a separate listener type for segrep.  I think we could extract PeerRecoveryTargetService.RecoveryListener and use it for segrep with a common exception type.  We will still need to perform the same actions that `RecoveryListener` now performs on success (publishing shardStarted) and failure (failing and removing the shard).

Reply:
Understood. I'll punt this to a future refactor